src/main/cpp/log_writer.cpp

```diff
@@ -121,10 +118,21 @@ void LogWriter::inspectMethod(const method_id methodId,
     frameLookup_(frame, jvmti_, *this);
 }
 
-void LogWriter::recordTraceStart(const jint numFrames, const map::HashType threadId) {
+void LogWriter::recordTraceStart(const jint numFrames, map::HashType envHash, const timespec &ts, ThreadBucket *info) {
+    map::HashType threadId = envHash;
+    char *threadName = (char*)"";
+
+    if (info) {
+        threadId = (map::HashType) info->tid;
+        threadName = info->name;
+    }
+
     output_.put(TRACE_START);
```

Comment:
For backward compatibility you could create a new record type, TRACE_WITH_TIME, and keep compatibility with old logs that way.